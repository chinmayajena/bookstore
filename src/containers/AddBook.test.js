import React from "react";
import { shallow, mount } from "enzyme";
import ConnectedAddBook, { AddBook } from "./AddBook.js";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

describe("<AddBook />", () => {
  let component;
  const addBookfn = jest.fn();
  let appHistory = [];
  beforeEach(() => {
    component = shallow(<AddBook addBook={addBookfn} history={appHistory} />);
  });

  it("renders the component", () => {
    expect(component.debug()).toMatchSnapshot();
    const wrapper = component.find("ErrorBoundary");
    expect(wrapper.length).toBe(1);
  });

  it("submits the form", () => {
    component.find("ReduxForm").simulate("submit", { preventDefault() {} });
    expect(addBookfn.mock.calls.length).toBe(1);
  });
});

describe("<ConnectedAddBook />", () => {
  const initialState = {};
  const mockStore = configureStore();
  const addBookfn = jest.fn();
  let store, container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = mount(
      <Provider store={store}>
        <ConnectedAddBook addBook={addBookfn} />
      </Provider>
    );
  });

  it("+++ render the connected component", () => {
    expect(container).toMatchSnapshot();
    expect(container.length).toEqual(1);
  });
});
