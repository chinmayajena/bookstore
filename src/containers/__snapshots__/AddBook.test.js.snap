// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddBook /> renders the component 1`] = `
"<ErrorBoundary>
  <ReduxForm onSubmit={[Function]} touchOnBlur={true} touchOnChange={false} persistentSubmitErrors={false} destroyOnUnmount={true} shouldAsyncValidate={[Function: defaultShouldAsyncValidate]} shouldValidate={[Function: defaultShouldValidate]} shouldError={[Function: defaultShouldError]} shouldWarn={[Function: defaultShouldWarn]} enableReinitialize={true} keepDirtyOnReinitialize={true} updateUnregisteredFields={true} getFormState={[Function: getFormState]} pure={true} forceUnregisterOnUnmount={false} form=\\"createBook\\" />
</ErrorBoundary>"
`;

exports[`<ConnectedAddBook /> +++ render the connected component 1`] = `ReactWrapper {}`;
